sdk/src/azure/iot/az_iot_pnp_client_twin.c

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/iot/az_iot_pnp_client.h>
+
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/internal/az_result_internal.h>
+
+static const az_span iot_hub_twin_desired = AZ_SPAN_LITERAL_FROM_STR("desired");
+static const az_span iot_hub_twin_desired_version = AZ_SPAN_LITERAL_FROM_STR("$version");
+static const az_span property_response_value_name = AZ_SPAN_LITERAL_FROM_STR("value");
+static const az_span property_ack_code_name = AZ_SPAN_LITERAL_FROM_STR("ac");
+static const az_span property_ack_version_name = AZ_SPAN_LITERAL_FROM_STR("av");
+static const az_span property_ack_description_name = AZ_SPAN_LITERAL_FROM_STR("ad");
+static const az_span component_property_label_name = AZ_SPAN_LITERAL_FROM_STR("__t");
+static const az_span component_property_label_value = AZ_SPAN_LITERAL_FROM_STR("c");
+
+AZ_NODISCARD az_result az_iot_pnp_client_twin_parse_received_topic(
+    az_iot_pnp_client const* client,
+    az_span received_topic,
+    az_iot_pnp_client_twin_response* out_response)
+{
+  _az_PRECONDITION_NOT_NULL(client);
+  _az_PRECONDITION_VALID_SPAN(received_topic, 1, false);
+  _az_PRECONDITION_NOT_NULL(out_response);
+
+  az_iot_hub_client_twin_response hub_twin_response;
+  _az_RETURN_IF_FAILED(az_iot_hub_client_twin_parse_received_topic(
+      &client->_internal.iot_hub_client, received_topic, &hub_twin_response));
+
+  out_response->request_id = hub_twin_response.request_id;
+  out_response->response_type
+      = (az_iot_pnp_client_twin_response_type)hub_twin_response.response_type;
+  out_response->status = hub_twin_response.status;
+  out_response->version = hub_twin_response.version;
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_iot_pnp_client_twin_property_begin_component(
```

Comment:
it's hard to tell if this is used to parse received twin or to build it from the API name.

az_iot_pnp_client_twin_report_property_begin_component?
az_iot_pnp_client_twin_property_builder_begin_component?

Reply:
`az_iot_pnp_client_twin_reported_property_begin_component()` doesn't sound too bad to me.

Reply:
> az_iot_pnp_client_twin_property_builder_begin_component

Using `builder` is interesting.

Reply:
update to use builder